### ISSUE TITLE
bam_get_qname returns null pointer in 1.10

### ISSUE THREAD

**Issue by Shians** [samtools/htslib#1080]
When using bam_get_qname on a `bam1_t`, I get a null pointer for the `char *` where in 1.9 I would get the sequence name. Snippet of C++ code looks like this

```c
    samFile *fp_in = hts_open(PRG_OPTS.bam.c_str(), "r");
    bam_hdr_t *bam_hdr = sam_hdr_read(fp_in);
    bam1_t *aln = bam_init1();

    // set up thread pool
    htsThreadPool p = {NULL, 0};
    p.pool = hts_tpool_init(PRG_OPTS.threads);
    hts_set_opt(fp_in, HTS_OPT_THREAD_POOL, &p);

    while (sam_read1(fp_in, bam_hdr, aln) > 0) {
        string read_name = bam_get_qname(aln);
    }
```

Code used to work in 1.9, but in 1.10 it would throw an error because strings cannot be constructed from null pointer.

**Comment by valeriuo** [comment]
Tried your code with HTSlib 1.9, 1.10.2 and current develop and it worked every time on a valid BAM file with 2 threads.
What compiler are you using? Also, can you provide a relevant subset of your test data?

**Comment by Shians** [comment]
Hmmm, it's strange I can reliably reproduce the problem inside my project, as well as fix it by rolling back to 1.9, but if I extract the code out by itself the problem goes away. Perhaps it's some strange interaction with other libraries I'm using in the larger project. I'll close this issue until I can reproduce the problem in a simpler setting.
